process/smartContract/builtInFunctions/esdtNFTCreateRoleTransfer.go

```diff
@@ -0,0 +1,175 @@
+	"github.com/ElrondNetwork/elrond-go/vm"
+)
+
+var _ process.BuiltinFunction = (*esdtNFTCreateRoleTransfer)(nil)
+
+type esdtNFTCreateRoleTransfer struct {
+	keyPrefix        []byte
+	marshalizer      marshal.Marshalizer
+	accounts         state.AccountsAdapter
+	shardCoordinator sharding.Coordinator
+}
+
+// NewESDTNFTCreateRoleTransfer returns the esdt NFT create role transfer built-in function component
+func NewESDTNFTCreateRoleTransfer(
+	marshalizer marshal.Marshalizer,
+	accounts state.AccountsAdapter,
+	shardCoordinator sharding.Coordinator,
+) (*esdtNFTCreateRoleTransfer, error) {
+	if check.IfNil(marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(accounts) {
+		return nil, process.ErrNilAccountsAdapter
+	}
+	if check.IfNil(shardCoordinator) {
+		return nil, process.ErrNilShardCoordinator
+	}
+
+	e := &esdtNFTCreateRoleTransfer{
+		keyPrefix:        []byte(core.ElrondProtectedKeyPrefix + core.ESDTKeyIdentifier),
+		marshalizer:      marshalizer,
+		accounts:         accounts,
+		shardCoordinator: shardCoordinator,
+	}
+
+	return e, nil
+}
+
+// SetNewGasConfig is called whenever gas cost is changed
+func (e *esdtNFTCreateRoleTransfer) SetNewGasConfig(_ *process.GasCost) {
+}
+
+// ProcessBuiltinFunction resolves ESDT create role transfer function call
+func (e *esdtNFTCreateRoleTransfer) ProcessBuiltinFunction(
+	acntSnd, acntDst state.UserAccountHandler,
+	vmInput *vmcommon.ContractCallInput,
+) (*vmcommon.VMOutput, error) {
+
+	err := checkBasicESDTArguments(vmInput)
+	if err != nil {
+		return nil, err
+	}
+	if !check.IfNil(acntSnd) {
+		return nil, process.ErrInvalidArguments
+	}
+	if check.IfNil(acntDst) {
+		return nil, process.ErrNilUserAccount
+	}
+
+	vmOutput := &vmcommon.VMOutput{ReturnCode: vmcommon.Ok}
+	if bytes.Equal(vmInput.CallerAddr, vm.ESDTSCAddress) {
+		outAcc, errExec := e.executeTransferNFTCreateChangeAtCurrentOwner(acntDst, vmInput)
+		if errExec != nil {
+			return nil, errExec
+		}
+		vmOutput.OutputAccounts[string(outAcc.Address)] = outAcc
+	} else {
+		err = e.executeTransferNFTCreateChangeAtNextOwner(acntDst, vmInput)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return vmOutput, nil
+}
+
+func (e *esdtNFTCreateRoleTransfer) executeTransferNFTCreateChangeAtCurrentOwner(
+	acntDst state.UserAccountHandler,
+	vmInput *vmcommon.ContractCallInput,
+) (*vmcommon.OutputAccount, error) {
+	if len(vmInput.Arguments) != 2 {
+		return nil, process.ErrInvalidArguments
+	}
+	if len(vmInput.Arguments[1]) != len(vmInput.CallerAddr) {
+		return nil, process.ErrInvalidArguments
+	}
+
+	tokenID := vmInput.Arguments[0]
+	nonce, err := getLatestNonce(acntDst, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = saveLatestNonce(acntDst, tokenID, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	destAddress := vmInput.Arguments[1]
+	if e.shardCoordinator.ComputeId(destAddress) == e.shardCoordinator.SelfId() {
+		newDestinationAcc, errLoad := e.accounts.LoadAccount(destAddress)
```

Comment:
maybe you can make a function for this code `saveLatestNonceOnDstAddress` or something like this

Reply:
would leave as it is.